web/src/main/java/org/fao/geonet/Geonetwork.java

```diff
@@ -558,6 +570,39 @@ private boolean checkDBWrite() {
         scheduledExecutorService.scheduleWithFixedDelay(DBHeartBeat, initialDelay, fixedDelay, TimeUnit.SECONDS);
     }
 
+    private void createLinkMonitor(final ResourceManager rm, final GeonetContext gc, String linkMonitorClass,
```

Comment:
Would this be better as a factory method on `LinkMonitorInterface`?  Some funky reflection going on here, maybe that might make it difficult.